lms/djangoapps/completion_api/views.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import absolute_import
+
+from rest_framework import viewsets
+from rest_framework.views import APIView
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from progress.models import StudentProgress
+
+from course_blocks.api import get_course_blocks
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from .serializers import StudentProgressSerializer
+
+
+IGNORE_CATEGORIES = {
+    # Structural types
+    'course',
+    'chapter',
+    'sequential',
+    'vertical',
+
+    # Non-completable types
+    'discussion-course',
+    'group-project',
+    'discussion-forum',
+    'eoc-journal',
+
+    # GP v2 categories
+    'gp-v2-project',
+    'gp-v2-activity',
+    'gp-v2-stage-basic',
+    'gp-v2-stage-completion',
+    'gp-v2-stage-submission',
+    'gp-v2-stage-team-evaluation',
+    'gp-v2-stage-peer-review',
+    'gp-v2-stage-evaluation-display',
+    'gp-v2-stage-grade-display',
+    'gp-v2-resource',
+    'gp-v2-video-resource',
+    'gp-v2-submission',
+    'gp-v2-peer-selector',
+    'gp-v2-group-selector',
+    'gp-v2-review-question',
+    'gp-v2-peer-assessment',
+    'gp-v2-group-assessment',
+    'gp-v2-static-submissions',
+    'gp-v2-static-grade-rubric',
+    'gp-v2-project-team',
+    'gp-v2-navigator',
+    'gp-v2-navigator-navigation',
+    'gp-v2-navigator-resources',
+    'gp-v2-navigator-submissions',
+    'gp-v2-navigator-ask-ta',
+    'gp-v2-navigator-private-discussion',
+}
+        
+       
+def annotate_progress(progress, collected_block_structure=None):
+    course_location = CourseOverview.load_from_module_store(progress.course_id).location
+    blocks = get_course_blocks(
+            progress.user,
+            course_location,
+            collected_block_structure=collected_block_structure
+    )
+    completable_blocks = {
+        block for block in blocks 
+        if blocks.get_xblock_field(block, 'category') not in IGNORE_CATEGORIES
+    }
+    progress.possible = float(len(completable_blocks))
+    progress.percent = int(round(100 * progress.completions / progress.possible))
+    return progress
+
+
+class CompletionDetailView(APIView):
```

Comment:
Request: In the docstring for this class, can you please link to the Completion API proposal wiki page, and explain that this is a transitional API implementation that will be replaced by the upstream version in the future?